core/src/main/java/com/yandex/yoctodb/util/mutable/impl/FoldedByteArrayIndexedList.java

```diff
@@ -0,0 +1,179 @@
+        long elementSize = 0;
+        for (UnsignedByteArray element : elements) {
+            elementSize += element.length();
+        }
+
+        return Integer.BYTES + // Element count in bytes
+                Integer.BYTES + // offsets count in bytes
+                sizeOfIndexOffsetValue *
+                        databaseDocumentsCount + // indexes offsets in bytes
+                Long.BYTES * offsets.size() + // offsets array size in bytes
+                elementSize; // Element array size in bytes
+    }
+
+    @Override
+    public void writeTo(
+            @NotNull final OutputStream os) throws IOException {
+        // elements count
+        os.write(Ints.toByteArray(databaseDocumentsCount));
+
+        // write offsets count!
+        // before writing indexes because
+        // you should to know how to deserialize it
+        os.write(Ints.toByteArray(offsets.size()));
+
+        List<Integer> offsetIndexes = new ArrayList<>();
+        int expectedDocument = 0;
+        for (Integer docId : docIdOffsetIndex.keySet()) {
+            while (expectedDocument < docId) {
+                offsetIndexes.add(0);
+                expectedDocument++;
+            }
+            offsetIndexes.add(docIdOffsetIndex.get(docId));
+            expectedDocument++;
+        }
+        while (expectedDocument < databaseDocumentsCount) {
+            offsetIndexes.add(0);
+            expectedDocument++;
+        }
+
+        // indexes of offsets - in correct Order
+        switch (sizeOfIndexOffsetValue) {
+            case (Byte.BYTES): {
+                // write every int to one byte
+                // The collection's iterator returns the elements
+                // in ascending order of the corresponding keys.
+                for (Integer offsetIndex : offsetIndexes) {
+                    os.write(oneByteFromInteger(offsetIndex));
+                }
+                break;
+            }
+            case (Short.BYTES): {
+                // write every int to two bytes
+                for (Integer offsetIndex : offsetIndexes) {
+                    os.write(twoBytesFromInteger(offsetIndex));
+                }
+                break;
+            }
+            case (Integer.BYTES): {
+                // write every int to four bytes
+                for (Integer offsetIndex : offsetIndexes) {
+                    os.write(Ints.toByteArray(offsetIndex));
+                }
+                break;
+            }
+        }
+
+        // offsets
+        for (Long offset : offsets) {
+            os.write(Longs.toByteArray(offset));
+        }
+
+        // elements
+        for (OutputStreamWritable e : elements) {
+            e.writeTo(os);
+        }
+    }
+
+    @Override
+    public String toString() {
+        return "FoldedByteArrayIndexedList{" +
+                "elementsCount=" + docIdOffsetIndex.size() +
+                '}';
+    }
+
+    private byte[] oneByteFromInteger(Integer data) {
```

Comment:
This method should become `static` and assert that `data` is in the `byte` range, don't you think so?